ui/pages/api/stripe/create-stripe-billing-portal-url.ts

```diff
@@ -0,0 +1,34 @@
+import handler from "server/api-handler";
+import prisma from "server/prisma";
+import stripe from "server/stripe";
+import { getRequestOrigin } from "server/get-request-origin";
+
+export default handler().get(async (req, res) => {
+    if (!req.user) throw new Error("You must be logged in");
+
+    const groupId = req.query.groupId;
+    // expect groupId to be a string
+    if (typeof groupId !== "string") throw new Error("groupId missing or not a string");
+
+    const origin = getRequestOrigin(req);
+    const group = await prisma.group.findUnique({ where: { id: groupId } });
+    if (!group.stripeCustomerId) throw new Error("Group has no customer id");
+
+    //check that user is admin of group
+    const groupMember = await prisma.groupMember.findFirst({
+        where: {
+            groupId,
+            userId: req.user.id
+        }
+    });
+    if (!groupMember.isAdmin) throw new Error("You need to be admin to access billing portal");
+
+    const { url } = await stripe.billingPortal.sessions.create({
+        customer: group.stripeCustomerId,
+        return_url: `${origin}/${group.slug}/settings/billing`,
```

Comment:
Can't redirect like this

Reply:
This also works in development, and what would be another strategy?

Reply:
The alternative would be like this https://github.com/cobudget/cobudget/blob/91ca4288d97de0f333d4853651cc7e843c3727ca/ui/pages/api/stripe/create-checkout-session.ts#L112-L116

Reply:
That saves it in a cookie that then gets used once you're back